docs/contributing.rst

```diff
@@ -2,12 +2,701 @@
+
+        $ git commit -m "Descriptive comment about what this commit does"
+
+   * Limiting the commit to file(s) changed for an atomic task would be very much
+     helpful in cases you need to review and maybe revert commits.
+
+   * Do not forget that `pre-commit hooks` will check your code changes at this
+     point. If some of those hooks fail, you will need to add those files again
+     and attempt to make the same commit again as the hooks will have changed those
+     file(s) to make them comply with the code formatting rules.
+
+   * A good practice is to run::
+
+        $ git status
+
+     after your commit to verify everything looks as expected.
+
+4. Push your commit(s) into the remote repository::
+
+    $ git push
+
+   or::
+
+    $ git push --set-upstream origin <your-branch-name>
+
+   if you haven't created a corresponding remote branch for it yet).
+
+.. note:: Remember that we are still not proposing to merge our work into the UXarray
+    GitHub repository, which will be described in the next subsection, `3.7. Pull
+    Requests`_.
+
+3.7. Pull Requests
+------------------
+
+Pull requests (PR) are GitHub's way for developers to propose and collaborate on changes
+to a repository. If you are interested in learning the foundations about pull requests,
+please refer to GitHub's `Pull requests <https://docs.github.com/en/pull-requests>`_
+documentation.
+
+Once you have completed making changes to your local copy of the UXarray repository,
+pushed them all, and are ready to have your changes merged into the repository on GitHub,
+you need to submit a PR asking the UXarray maintainers to consider your merge request.
+
+The merge will occur between your personal branch that should have all of your commits
+from local in the GitHub repository (either in your fork or in the actual UXarray repo)
+and the :code:`main` branch, if not other, in the UXarray's GitHub repository. There
+might be some exceptions to this generic case, which can always be learned through
+reading or discussed with the maintainers and community.
+
+.. note::
+    While pull requests are supposed to be based on changes that are ready to be tested,
+    reviewed, and eventually be merged to repositories; there is an exception to this:
+    Draft Pull Requests. We encourage Draft PRs if you just want to just start a
+    conversation about your code that isn’t in any state to be judged, and get feedback
+    as well as some guidance from others. Please read this GitHub `blog
+    <https://github.blog/2019-02-14-introducing-draft-pull-requests/>`_ about draft PRs.
+
+Please refer to Github's `Creating a pull request
+<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_
+guide for the instructions. We are especially avoiding to detail these instructions
+here as there are manly referrals to GitHub's graphical user interface, which might be
```

Comment:
```suggestion
here as they are mainly referrals to GitHub's graphical user interface, which might be
```
Depending on the intent of this sentence, this could also be changed to "there are many referrals..." instead.

Reply:
Done